pegasus/router.rb

```diff
@@ -383,6 +384,36 @@ def resolve_template(subdir, extnames, uri)
       nil
     end
 
+    # Scans the filesystem and finds all documents served by Pegasus CMS.
+    # @return [Array<Hash<Symbol, String>] An array of :site, :uri hash entries for all found documents.
+    def all_documents
+      dirs = Dir.children(content_dir).select {|file| Dir.exist?(content_dir(file))}
+      dirs.map do |site|
+        (settings.template_extnames - ['.fetch']).map do |ext|
+          site_glob = site_sub = content_dir(site, 'public')
+
+          if site == 'hourofcode.com'
+            # hourofcode.com has custom logic to include
```

Comment:
does this also include that hourofcode.com inherits from code.org pages?

Reply:
hourofcode.com doesn't have any `"base"` set in its [`config.json`](https://github.com/code-dot-org/code-dot-org/blob/staging/pegasus/sites.v3/hourofcode.com/config.json) that would inherit page content - did you mean csedweek.org? https://github.com/code-dot-org/code-dot-org/blob/b1204544aab056f4f485a3b5d5b62b56ac21c9ae/pegasus/sites.v3/csedweek.org/config.json#L2

In any case, the 'base' inheritance makes the same document available across multiple sites. For this first pass I only wanted to make sure each document on the filesystem was rendered at least once, rather than render each document a separate time for each site inheriting it. (It would be possible to extend this in a future PR for more thorough coverage across all routes.)

Reply:
This makes sense. I think we care more about validity of pages than where they get displayed right now.